services/src/main/scala/cromwell/services/metadata/MetadataService.scala

```diff
@@ -126,13 +128,27 @@ object MetadataService {
       List(MetadataEvent(metadataKey, MetadataValue(value)))
   }
 
-  def throwableToMetadataEvents(metadataKey: MetadataKey, t: Throwable): List[MetadataEvent] = {
+  def throwableToMetadataEvents(metadataKey: MetadataKey, t: Throwable, failureIndex: Int = Random.nextInt(Int.MaxValue)): List[MetadataEvent] = {
+    val emptyCauseList = List(MetadataEvent.empty(metadataKey.copy(key = metadataKey.key + s"[$failureIndex]:causedBy[]")))
+
     t match {
       case aggregation: ThrowableAggregation =>
-        aggregation.errorMessages.toList map { message => MetadataEvent(metadataKey.copy(key = s"${metadataKey.key}:message"), MetadataValue(s"${aggregation.exceptionContext}: $message")) }
+        val message = List(MetadataEvent(metadataKey.copy(key = s"${metadataKey.key}[$failureIndex]:message"), MetadataValue(aggregation.exceptionContext)))
+        val indexedCauses = aggregation.throwables.toList.zipWithIndex
+        val indexedCauseEvents = if (indexedCauses.nonEmpty) {
+          indexedCauses flatMap { case (cause, index) =>
+            val causeKey = metadataKey.copy(key = metadataKey.key + s"[$failureIndex]:causedBy")
+            throwableToMetadataEvents(causeKey, cause, index)
+          }
+        } else {
+          emptyCauseList
+        }
+
+        message ++ indexedCauseEvents
       case other =>
-        val message = List(MetadataEvent(metadataKey.copy(key = s"${metadataKey.key}:message"), MetadataValue(t.getMessage)))
-        val cause = Option(t.getCause) map { cause => throwableToMetadataEvents(metadataKey.copy(key = s"${metadataKey.key}:causedBy"), cause) } getOrElse List.empty
+        val message = List(MetadataEvent(metadataKey.copy(key = s"${metadataKey.key}[$failureIndex]:message"), MetadataValue(t.getMessage)))
+        val causeKey = metadataKey.copy(key = metadataKey.key + s"[$failureIndex]:causedBy")
```

Comment:
`${metadataKey.key}[$failureIndex]:` sure gets used a lot here, can this be pulled out to a val?